pkg/resource/gcs.go

```diff
@@ -0,0 +1,224 @@
+package resource
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/url"
+	"os"
+	"sync"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"google.golang.org/api/iterator"
+
+	"github.com/leaktk/scanner/pkg/logger"
+	"github.com/leaktk/scanner/pkg/response"
+)
+
+var getClientMu sync.Mutex
+var storageClient *storage.Client
+
+// newStorageClient returns a usable storage client for resources. Since
+// storage.Client is safe across multiple threads, it returns the same
+// client each time
+func newStorageClient() (*storage.Client, error) {
```

Comment:
It seems like this would be a good use of [`sync.Once`](https://victoriametrics.com/blog/go-sync-once/).